### ISSUE TITLE
Multi-module Gradle doesn't resolve properties set in parent build file

### ISSUE THREAD

**Issue by kdubb** [dekorateio/dekorate#1086]
In a multi-module Gradle project with properties (e.g. version) set for all projects in the root `build.gradle.kts`, dekorate doesn't resolve those properties and emits a warning.

Example `/build.gradle.kts`:
```
allprojects {
  version = "1.0"
}
```

This will emit the warning `Could not detect project version. Using 'latest'.`.

**Comment by iocanel** [comment]
I am not really sure how much we can do about it, given that purposefully we avoid bring in build tool dependencies and we are parsing files ourselves. The only possible solution I could think of is navigating the directory hierarchy from module root to project root and try to detect global settings.

**Comment by kdubb** [comment]
Is this not what is done for maven?

**Comment by iocanel** [comment]
No, AFAIR for maven the version is read from the parent element.

**Comment by kdubb** [comment]
@iocanel There's any easy fix for this. Require that users define the version in `gradle.properties`; which is an accepted practice. Then, search up the tree until you hit a `gradle.properties`.

Currently the code is only looking in the current project directory, but for multi-module project it needs to search up until the root project directory. 

The best solution would gather all the `gradle.properties` up until the root and then apply them in reverse order to mimic the overriding that happens.
